test/core/networking/websockets/middleware/encryption_middleware_test.dart

```diff
@@ -0,0 +1,229 @@
+    await receivedNormalMessage.awaitValue(Duration(seconds: 10), raiseOnTimeout: true);
+
+    // Send crypto message
+    session.send('\$ANYTHING-2');
+
+    // Check exception raised and connection closed
+    await raisedException.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+    await closedSession.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+
+    receivedNormalMessage.expect(equals("ANYTHING-1"));
+    receivedCryptoMessage.expect(isNull);
+    closedSession.expect(isTrue);
+    raisedException.expect(
+      isA<HandshakeException>().having(
+        (HandshakeException e) => e.message,
+        'message',
+        equals('Received crypto message after secure connection was established'),
+      ),
+    );
+
+    expect(session.storage['crypto_provider'], isNull);
+  });
+
+  test('EncryptionMiddleware throws on invalid crypto message', () async {
+    final raisedException = VHook<Object?>(null);
+    final closedSession = VHook<bool?>(null);
+
+    final List<MiddlewareFunc> middleware = [encryptionMiddleware];
+
+    // Create server to accept connection
+    final server = await openWebServer();
+    server.wsMiddleware = middleware;
+    server.wsOnMessage = (ServerSession session, dynamic msg) async {};
+    server.wsOnError = (ServerSession session, Object exception) {
+      raisedException.set(exception);
+    };
+
+    // Connect to the server
+    final client = await WebSocket.connect(getUri(server, 'ws').toString());
+    client.listen((dynamic _) {}, onDone: () {
+      closedSession.set(true);
+    });
+
+    // Send invalid crypto message
+    client.add('\$ANYTHING');
+
+    // Check exception raised and connection closed
+    await raisedException.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+    await closedSession.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
+
+    raisedException.expect(
+      isA<HandshakeException>()
+          .having(
+            (HandshakeException e) => e.message,
+            'message',
+            startsWith("Invalid crypto message: '"),
+          )
+          .having(
+            (HandshakeException e) => e.message,
+            'message',
+            endsWith("'"),
+          ),
+    );
+  });
+
+  test('EncryptionMiddleware throws on invalid public key', () async {
+    final raisedException = VHook<Object?>(null);
+    final closedSession = VHook<bool?>(null);
+
+    final List<MiddlewareFunc> middleware = [encryptionMiddleware];
+
+    // Create server to accept connection
+    final server = await openWebServer();
+    server.wsMiddleware = middleware;
+    server.wsOnMessage = (ServerSession session, dynamic msg) async {};
+    server.wsOnError = (ServerSession session, Object exception) {
+      raisedException.set(exception);
+    };
+
+    // Connect to the server
+    final client = await WebSocket.connect(getUri(server, 'ws').toString());
+    client.listen((dynamic _) {}, onDone: () {
```

Comment:
```suggestion
    client.listen((_) {}, onDone: () {
```

Reply:
Same reason as before, don’t want to change